server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -859,7 +859,7 @@ public List<Reference> collectSchemaUpdates(IndexItem item) throws IOException {
         Object[] values = item.insertValues();
         for (int i = 0; i < values.length; i++) {
             Reference reference = columns.get(i);
-            Object value = valueForInsert(reference.valueType(), values[i]);
+            Object value = values[i];
```

Comment:
As far as I can tell `valueForInsert` is not relevant to detect schema updates. The call happens later when indexing and removing it here avoids copy operations for array and objects.

(valueForInsert also called `sanitizeValue` already - which is called again a few lines below - so we always did that twice)